client/packages/system/src/Vaccination/api/useVaccination.ts

```diff
@@ -111,7 +111,7 @@ export function useVaccination({
     isDirty: Object.keys(patch).length > 0,
     updateDraft: (update: Partial<VaccinationDraft>) =>
       setPatch({ ...patch, ...update }),
-    create: insert,
+    create: encounterId ? insert : () => {},
```

Comment:
Bit ugly, but need to make sure the mutation can't be called when there is no `encounterId` (and can't conditionally call the `useInsert` hook, so passing a dummy value to it.)

Reply:
What if we let useInsert accept an optional encounterId, and do an early return from the mutationFn there? Same deal but feels a little more type safe - if the id is there, we know it's actually populated

Reply:
5e61809fd8